examples/bridge-app/linux/include/Clusters.h

```diff
@@ -0,0 +1,294 @@
+// functionality.
+template <chip::AttributeId id, EmberAfAttributeMask mask, typename Type>
+struct Attribute : public Type
+{
+    static constexpr chip::AttributeId kId    = id;
+    static constexpr EmberAfClusterMask kMask = mask | ZAP_ATTRIBUTE_MASK(EXTERNAL_STORAGE);
+
+    using Type::Type;
+    using Type::operator=;
+};
+
+struct CommonCluster;
+typedef std::function<CHIP_ERROR(CommonCluster *, const chip::app::ConcreteDataAttributePath &, chip::app::AttributeValueDecoder &)>
+    PropagateWriteCB;
+
+// This is the base type of all generated clusters, providing the backend access to
+// implementation details.
+struct CommonCluster : public ClusterImpl
+{
+    void SetEndpointId(chip::EndpointId id) override;
+    chip::EndpointId GetEndpointId() const;
+
+    void SetCallback(PropagateWriteCB * cb);
+
+    bool active() const;
+
+    virtual CHIP_ERROR WriteFromBridge(const chip::app::ConcreteDataAttributePath & aPath,
+                                       chip::app::AttributeValueDecoder & aDecoder) = 0;
+
+    CHIP_ERROR ForwardWriteToBridge(const chip::app::ConcreteDataAttributePath & aPath,
+                                    chip::app::AttributeValueDecoder & aDecoder);
+
+    template <typename T, typename U>
+    void Push(T & attr, U v)
+    {
+        attr = v;
+        OnUpdated(T::kId);
+    }
+    bool Push(chip::AttributeId attr, chip::TLV::TLVReader & reader);
+
+    void OnUpdated(chip::AttributeId attr);
+    template <typename T>
+    void OnUpdated(const T & attr)
+    {
+        OnUpdated(T::kId);
+    }
+
+protected:
+    chip::EndpointId mEndpoint   = 0xFFFF;
+    PropagateWriteCB * mCallback = nullptr;
+};
+
+struct CommonAttributeAccessInterface : public chip::app::AttributeAccessInterface
+{
+    using chip::app::AttributeAccessInterface::AttributeAccessInterface;
+    static CommonCluster * FindCluster(const chip::app::ConcreteClusterPath & path);
+};
+
+#include "bridge/BridgeClustersImpl.h"
+
+namespace clusters {
+struct BridgedDeviceBasicCluster : public CommonCluster
+{
+    static constexpr chip::ClusterId kClusterId = ZCL_BRIDGED_DEVICE_BASIC_CLUSTER_ID;
+
+    BridgedDeviceBasicCluster();
+
+    chip::ClusterId GetClusterId() override { return kClusterId; }
+    CHIP_ERROR WriteFromBridge(const chip::app::ConcreteDataAttributePath & aPath,
+                               chip::app::AttributeValueDecoder & aDecoder) override;
+
+    template <typename T>
+    void AddAllAttributes(T * list)
+    {
+        list->Add(mReachable);
+    }
+
+    chip::Span<const EmberAfAttributeMetadata> GetAllAttributes() override
+    {
+        static constexpr EmberAfAttributeMetadata kAllAttributes[] = {
+            { ZCL_REACHABLE_ATTRIBUTE_ID, ZCL_BOOLEAN_ATTRIBUTE_TYPE, 1, 0, 0u },
+        };
+        return chip::Span<const EmberAfAttributeMetadata>(kAllAttributes);
+    }
+
+    Attribute<ZCL_REACHABLE_ATTRIBUTE_ID, 0, PrimitiveType<uint8_t, 1, ZCL_BOOLEAN_ATTRIBUTE_TYPE>> mReachable;
```

Comment:
Add it or not add it to the list of attributes for that cluster. Presumably any access to an attribute not in that list is already an error so doing dummy accesses should be acceptable. Otherwise we'll have to store a list of enabled attributes in here

Reply:
But this is code-generated right? How would an application developer using this code-generated class be able to impart that level of customization?

Reply:
Yes, the default list adds all attributes, but if a specific app wanted to support fewer attributes then it could when building the device, adding only the attributes it supports. Different devices can support different sets of attributes for the same cluster type as well. Space would still be allocated for every attribute which is slightly wasteful, but it would work. Alternatively, the IDL could be edited to remove unsupported attributes entirely